buildSrc/src/main/java/org/springframework/boot/build/DocumentConventions.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2012-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.boot.build;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+
+import org.gradle.api.GradleException;
+import org.gradle.api.Project;
+import org.gradle.api.file.CopySpec;
+import org.gradle.api.resources.TextResourceFactory;
+import org.gradle.jvm.tasks.Jar;
+
+import org.springframework.util.FileCopyUtils;
+
+/**
+ * Class to support the generation of NOTICE.txt and LICENSE.txt files as a result of the
+ * {@link Jar} task being configured in {@link JavaConventions} class.
+ *
+ * @author Andy Wilkinson
+ * @author Christoph Dreis
+ * @author Mike Smithson
+ */
+class DocumentConventions {
```

Comment:
I'd prefer to keep these in the same class as the other Java-related conventions.

Reply:
OK, I inlined all of the methods in DocumentConventions into JavaConventions.